haas/client/auth.py

```diff
@@ -0,0 +1,32 @@
+import requests
```

Comment:
I don't like the way you've structured the auth stuff at all. Here's what I would do:

1. Take the HTTPClient classes as they are on master (i.e. without any of the modifications you've done), and move them to the client library.
2. Rather than having the session parameter to the Client class, just accept an HTTPClient.

Then you can just delete this file.

The auth stuff is already done and in a usable form on master; you should just use it as is.